pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -167,8 +212,10 @@ func (r *ReconcileCheInstallation) ensureCheNamespace(logger logr.Logger, cheIns
 				return false, err
 			}
 			if ns.Status.Phase != v1.NamespaceActive {
-				logger.Info("Namespace is not in active state", "namespace", ns.Name, "phase", ns.Status.Phase)
-				return true, nil // requeue until the namespace is active
+				logger.Info("Namespace is not in active state - deleting remaining CheCluster resource", "namespace", ns.Name, "phase", ns.Status.Phase)
+				// return 'true' in any case, as we don't want to continue with the current reconciliation loop
+				_, err = r.ensureCheClusterDeletion(logger, cheInstallation)
```

Comment:
do we need this?

Reply:
ok, now I understand:

> Also, when the namespace is in 'Terminating' state (if
an admin deleted it manually), we delete the CheCluster
for the same reason.

but isn't that trying to fix someone's else error? I mean, from now on it shouldn't happen that the namespace is in `Terminating` state because we didn't do the proper cleanup.
In addition, if the operator is uninstalled and the `CheCluster` is not removed (because it contains finalizer) then calling another `delete` for that resource will not help anyhow. Or am I missing something?

Reply:
well, yes, it's aimed at making sure that the namespace is not stuck in `Terminating` phase if an admin deletes the namespace (which will be recreated by our operator afterwards). It does not cover all cases for sure, but at least this one will work.

Reply:
ok, I got your point, and you're right indeed! I reverted this change in f40f0f9